src/peft/tuners/lora/gptq.py

```diff
@@ -77,10 +83,19 @@ def forward(self, x: torch.Tensor):
                 expected_dtype = result.dtype
                 x = self._cast_input_dtype(x, lora_A.weight.dtype)
 
-            output = lora_B(lora_A(dropout(x)))
+            # lora_dropout float value is not stored so we need to check for cls
+            if isinstance(dropout, torch.nn.Dropout):
+                output = lora_B(lora_A(dropout(x)))
+            else:
+                # dropout == Identity which is no-op if lora_dropout == 0.0
+                output = lora_B(lora_A(x))
```

Comment:
@SunMarc  Another micro optimization no-op skip. `lora_dropout: float` is not stored so if `0.0` the `update_layers` assign an `Identity` module which does nothing no-op.  Only do dropout if dropout is activated `lora_dropout: float > 0.0`

Reply:
Just did a benchmark and it does not affect runtime. Looke pytorch is able to optimize it away. Will remove.